rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java

```diff
@@ -298,21 +286,53 @@ public ResponseEntity<CollectionModel<EntityModel<Project>>> getProjectsForUser(
         } else {
             if (isSearchByName) {
                 sw360Projects.addAll(projectService.searchProjectByName(params.get("name"), sw360User));
-            } else if (isSearchByGroup) {
-                sw360Projects.addAll(projectService.searchProjectByGroup(group, sw360User));
-            } else if (isSearchByTag) {
-                sw360Projects.addAll(projectService.searchProjectByTag(params.get("tag"), sw360User));
-            } else if (isSearchByType) {
-                sw360Projects.addAll(projectService.searchProjectByType(projectType, sw360User));
             } else {
+                isAllProjectAdded=true;
                 sw360Projects.addAll(projectService.getProjectsForUser(sw360User, pageable));
+            }
+            Map<String, Set<String>> restrictions = getFilterMap(tag, projectType, group, version, projectResponsible, projectState, projectClearingState,
+                    additionalData);
+            if (!restrictions.isEmpty()) {
+                sw360Projects = new ArrayList<>(sw360Projects.stream()
+                        .filter(filterProjectMap(restrictions)).toList());
+            }else if(isAllProjectAdded){
                 isNoFilter = true;
             }
         }
         return getProjectResponse(pageable, projectType, group, tag, allDetails, luceneSearch, request, sw360User,
                 mapOfProjects, isSearchByName, sw360Projects, isNoFilter);
```

Comment:
`getProjectResponse()` no longer requires filter of `projectType`, `group`, etc. as it is already done by `filterProjectMap()` and `refineSearch()`.